src/Sentry.Extensions.AI/Extensions/SentryAIExtensions.cs

```diff
@@ -0,0 +1,59 @@
+using Sentry.Extensions.AI;
+using Sentry.Infrastructure;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Extensions to instrument Microsoft.Extensions.AI builders with Sentry
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class SentryAIExtensions
+{
+    /// <summary>
+    /// Wrap tool calls specified in <see cref="ChatOptions"/> with Sentry agent instrumentation
+    /// </summary>
+    /// <remarks>
+    /// This API is experimental, and it may change in the future.
+    /// </remarks>
+    /// <param name="options">The <see cref="ChatOptions"/> that contains the <see cref="AIFunction"/> to instrument</param>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public static ChatOptions AddSentryToolInstrumentation(this ChatOptions options)
+    {
+        if (options.Tools is { Count: 0 })
```

Comment:
```suggestion
        if (options.Tools is not { Count: > 0 })
```

I think the logic got broken in the refactoring... the above basically reads:
> If `options.Tools is not a non-null object with a `Count` greater than zero

The logic you have currently fails to match null values... it only matches a non-null object with a count of zero.

Reply:
Fixed  in https://github.com/getsentry/sentry-dotnet/pull/4657/commits/ac9f7ceb1f22f3e0d325f7314f830c596cbb4018